auv_navigation/auv_navigation/scripts/follow_path_action_server.py

```diff
@@ -0,0 +1,233 @@
+#!/usr/bin/env python3
+
+from actionlib import SimpleActionServer
+import rospy
+import actionlib
+import numpy as np
+from nav_msgs.msg import Path
+from geometry_msgs.msg import Twist
+from std_msgs.msg import Bool
+from auv_msgs.msg import FollowPathAction, FollowPathFeedback, FollowPathResult, FollowPathGoal
+from auv_msgs.srv import AlignFrame, AlignFrameResponse
+import tf2_ros
+
+from navigation_utils.path_utils import (
+    create_path_from_frame, get_current_pose, 
+    calculate_carrot_pose, broadcast_carrot_frame
+)
+from navigation_utils.align_frame_utils import AlignFrameController
+
+class FollowPathActionServer:
+    def __init__(self):
+        rospy.init_node('follow_path_action_server')
+        self.tf_buffer = tf2_ros.Buffer()
+        self.tf_listener = tf2_ros.TransformListener(self.tf_buffer)
+        self.tf_broadcaster = tf2_ros.TransformBroadcaster()
+        self.carrot_distance = rospy.get_param('~carrot_distance', 1.0)
```

Comment:
is this the distance where the carrot is moved to the next pose? if so `dynamic_target_distance_threshold` is more appropriate?